migration-tools/migrate-rosdistro.py

```diff
@@ -80,13 +80,15 @@ def write_tracks_file(tracks, commit_msg=None):
 repositories_to_retry = []
 for repo_name, repo_data in sorted(source_distribution.repositories.items()):
     if repo_name not in dest_distribution.repositories:
-        new_repositories.append(repo_name)
         dest_repo_data = copy.deepcopy(repo_data)
-        release_tag = dest_repo_data.release_repository.tags['release']
-        release_tag = release_tag.replace(args.source,args.dest)
-        dest_repo_data.release_repository.tags['release'] = release_tag
+        if dest_repo_data.release_repository:
+            new_repositories.append(repo_name)
+            release_tag = dest_repo_data.release_repository.tags['release']
```

Comment:
These tags should be checked for presence before being modified. They might not exist too.

Reply:
The `release` tag is required according to REP-141 https://www.ros.org/reps/rep-0141.html#distribution-file so if we've got a release repo this should be there. With that being said, if you think this merits a defensive check, I'll add one.